### PR TITLE
Add server parameter for WebDriverAgent local port

### PR DESCRIPTION
**Pull request by domuro**
## Proposed changes

Enable specifying the WebDriverAgent port as a server parameter, so multiple Appium XCUI tests can be run concurrently.
## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://docs.google.com/forms/d/1lOfXRw_0VCk7gYzjj4WLetGu7yelDVo5LWh0z7pGftE/viewform)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by imurchie** [comment]
This is a desired capability. Why do you want it to be a server flag?

**Comment by yaphatak** [comment]
@imurchie I think it will be better to have this so that people can launch a node per device with different port. This is something similar to what we have for `--chromedriver-port` or `selendroid`

File: lib/parser.js [hunk 1]
```diff
@@ -351,6 +351,15 @@ const args = [
     help: '(IOS-only) Local port used for communication with ios-webkit-debug-proxy'
   }],
 
+  [['--web-driver-agent-port'], {
```
Comment (imurchie) [inline lib/parser.js]: I would prefer `webdriveragent-port`.

File: lib/parser.js [hunk 2]
```diff
@@ -351,6 +351,15 @@ const args = [
...
+    defaultValue: 8100,
+    dest: 'wdaLocalPort',
+    required: false,
+    type: 'int',
+    example: "8100",
+    help: '(IOS-only) Local port used for communication with WebDriverAgent'
```
Comment (imurchie) [inline lib/parser.js]: This should include info that it is XCUITest only, not just iOS only.

**Review by imurchie — CHANGES_REQUESTED** [review]
Thanks for this! A couple of changes, but otherwise good.

**Review by imurchie — APPROVED** [review]

**Comment by imurchie** [comment]
Awesome! Thanks for this!

**Pull request merged**